app/models/post.rb

```diff
@@ -1,5 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
   has_many :likes
   has_many :comments
+
+  def increment_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] I would suggest using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` to call the #increment_post_counter method every time a new post has been saved to database, then make the incremet_post_counter method private. See the example below 👇 :
```ruby
after_save :increment_post_counter

  private

  def increment_post_counter
    author.increment!(:posts_counter)
  end
```